src/v/cluster/types.h

```diff
@@ -2594,6 +2594,54 @@ struct cancel_partition_movements_reply
     std::vector<move_cancellation_result> partition_results;
 };
 
+/**
+ * Broker state transitions are coordinated centrally as opposite to
+ * configuration which change is requested by the described node itself. Broker
+ * state represents centrally managed node properties. The difference between
+ * broker state and configuration is that the configuration change is made on
+ * the node while state changes are managed by the cluster controller.
+ */
+class broker_state : public serde::envelope<broker_state, serde::version<0>> {
```

Comment:
i guess now we are supposed to do serde::compat_version explicitly?

Reply:
oh, right, fixing